js/gravityforms-mfi-multifile.js

```diff
@@ -0,0 +1,290 @@
+/*
+Copyright 2009-2016 Rocketgenius, Inc.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see http://www.gnu.org/licenses.
+*/
+
+//----------------------------------------
+//------ MULTIFILE UPLOAD FUNCTIONS ------
+//----------------------------------------
+
+(function (gfMultiFileUploader, $) {
+    gfMultiFileUploader.uploaders = {};
+    var strings = typeof gform_gravityforms != 'undefined' ? gform_gravityforms.strings : {};
+    var imagesUrl = typeof gform_gravityforms != 'undefined' ? gform_gravityforms.vars.images_url : "";
+
+    $(document).unbind('gform_post_render');
+    $(document).bind('gform_post_render', function(e, formID){
+
+        $("form#gform_" + formID + " .gform_fileupload_multifile").each(function(){
+            setup(this);
+        });
+        var $form = $("form#gform_" + formID);
+        if($form.length > 0){
+            $form.submit(function(){
+                var pendingUploads = false;
+                $.each(gfMultiFileUploader.uploaders, function(i, uploader){
+                    if(uploader.total.queued>0){
+                        pendingUploads = true;
+                        return false;
+                    }
+                });
+                if(pendingUploads){
+                    alert(strings.currently_uploading);
+                    window["gf_submitting_" + formID] = false;
+                    $('#gform_ajax_spinner_' + formID).remove();
+                    return false;
+                }
+            });
+        }
+
+    });
+
+    gfMultiFileUploader.setup = function (uploadElement){
+        setup( uploadElement );
+    };
+
+    function setup(uploadElement){
+        var settings = $(uploadElement).data('settings');
+
+        var uploader = new plupload.Uploader(settings);
+        formID = uploader.settings.multipart_params.random_id;
```

Comment:
Here we assign the formID used througout this file to the form's random id, injected here (https://github.com/tyxla/Gravity-Forms-Multiple-Form-Instances/pull/24/files#diff-7d573d66d37a99e5734d905449d49972R100) into the multifile upload field multipart_param.
It could also be added to another attribute, to avoid sending the value to the server while adding files, but i let it this way to simplify the code.